### PR TITLE
core-programming solution

### PR DESCRIPTION
**Pull request by kumar025**
Attempted to solve 7 questions out of 9. Will try to solve remaining as well.

### CODE REVIEW

File: Solutions/AlicePrimeStrings.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+		}
+	}
+
+	/**
+	 * Checks whether number is prime or not
+	 * 
+	 * @param number to be checked for prime
+	 * @return true if prime false otherwise
+	 */
+	private static boolean isPrime(int number) {
```
Comment (bsmaheshkumar) [inline Solutions/AlicePrimeStrings.java]: Why is this method a static?

File: Test_Cases/JohnGemTest.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	@After
+	public void tearDown() {
+		johnGemInstance = null;
+	}
+
+	/**
+	 * Test count gem positive.
+	 */
+	@Test
+	public void testFindNumberofGemElement() {
```
Comment (bsmaheshkumar) [inline Test_Cases/JohnGemTest.java:41]: Can you add more unit test case?
Also share the screen shot or unit test coverage?

File: Solutions/NonReducible.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+	public NonReducible findReducedStringAndOperationCount(String stringToBeReduced) {
+		this.operationCount = 0;
+		String reducibleString = "";
+		NonReducible nonReducible = new NonReducible();
+
+		if (stringToBeReduced != null && !stringToBeReduced.isEmpty()) {
+			for (int i = 0; i < stringToBeReduced.length() - 1;) {
+				char[] inputCharArr = stringToBeReduced.toCharArray();
+
+				if (inputCharArr[i] == inputCharArr[i + 1]) {
```
Comment (bsmaheshkumar) [inline Solutions/NonReducible.java:60]: What will happen if I give a single char has input?

File: Solutions/BrianCarName.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+	 * This method checks whether Brian's car name is OK or not. It returns OK
+	 * only if satisfies the following conditions Name must Consist exactly
+	 * three distinct characters All distinct characters must occur same number
+	 * of times i.e. if C1 occurred 2 times, C2 and C3 must also occur 2 times.
+	 * 
+	 * @param str
+	 * @return true or false
+	 */
+	public String checkBrianCarName(String inputString) {
+		String isBrianCarNameOk = "OK";
```
Comment (bsmaheshkumar) [inline Solutions/BrianCarName.java]: What other name can we give for this variable? isBrianCarNameOk

**Review by bsmaheshkumar — CHANGES_REQUESTED** [review]
Few minor comments and clarifications.

File: Solutions/AlicePrimeStrings.java (hunk 1 above)
Comment (kumar025) [reply Solutions/AlicePrimeStrings.java]: static is not required here. Should be removed.

File: Solutions/BrianCarName.java (hunk 4 above)
Comment (kumar025) [reply Solutions/BrianCarName.java]: isBrianCarNameOk is not a proper name for String type should be changed to carNameStatus

File: Solutions/NonReducible.java (hunk 3 above)
Comment (kumar025) [reply Solutions/NonReducible.java:60]: Ideally same single char should be returned. Handled this case and modified the code accordingly.

File: Test_Cases/JohnGemTest.java (hunk 2 above)
Comment (kumar025) [reply Test_Cases/JohnGemTest.java:41]: Added few more unit test cases.

File: Solutions/NonReducible.java (hunk 3 above)
Comment (bsmaheshkumar) [reply Solutions/NonReducible.java:60]: will it not crash?

**Review by bsmaheshkumar — COMMENTED** [review]
Looks good to me thanks.

**Comment by bsmaheshkumar** [comment]
Thanks for this...
I shall close the request.
Now its left to manager's.

**Pull request closed without merging**